File: frontend/src/types/notification.ts
export type NotificationType =
    | "opportunity_created"
    | "kyc_completed"
    | "kyc_failed"
    | "status_changed"
    | "deal_value_changed"
    | "opportunity_assigned"
    | "meeting_scheduled"
    | "meeting_reminder"
    | "proposal_reminder"
    | string;

export interface Notification {
    id: string;
    user_id: string;
    opportunity_id: string | null;
    type: NotificationType;
    title: string;
    message: string;
    link_url?: string | null;
    is_read: boolean;
    created_at: string;
}

export interface NotificationListResponse {
    items: Notification[];
    total: number;
    unread_count: number;
}

export interface UnreadCountResponse {
    unread_count: number;
}

File: backend/app/models/notification.py
import uuid
from datetime import datetime

from sqlalchemy import Column, String, DateTime, Boolean, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.core.database import Base


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    opportunity_id = Column(
        UUID(as_uuid=True), ForeignKey("opportunities.id", ondelete="CASCADE"), nullable=True
    )
    type = Column(String(50), nullable=False)  # opportunity_created, kyc_completed, status_changed, meeting_reminder, follow_up
    title = Column(String(255), nullable=False)
    message = Column(Text, nullable=True)
    is_read = Column(Boolean, default=False)
    metadata_json = Column("metadata", Text, nullable=True)  # JSON string for extra data
    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    user = relationship("User", back_populates="notifications")
    opportunity = relationship("Opportunity", back_populates="notifications")